src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp

```diff
@@ -209,6 +215,222 @@ Scalar<DataType> time_deriv_of_lapse(
   return dt_lapse;
 }
 
+template <size_t SpatialDim, typename Frame, typename DataType>
+void spatial_deriv_of_shift(
+    const gsl::not_null<tnsr::iJ<DataType, SpatialDim, Frame>*> deriv_shift,
+    const Scalar<DataType>& lapse,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept {
+  for (size_t i = 0; i < SpatialDim; ++i) {
```

Comment:
you need to initialize `deriv_shift`  (you can't assume this is called by the return-by-value function).  Instead of setting the whole thing to zero, you should factor out of the loop the first term and set `deriv_shift` to this value, and then continue to add the rest of the pieces.  Also you should first have an  `if (UNLIKELY()` clause, in which you check the size of `deriv_shift` and reset it to the size of the passed in arguments.

Reply:
@kidder  - changed everywhere

Reply:
You're missing the `UNLIKELY` part

Reply:
I've added an unlikely branch predictor with __builtin_expect, but, if the general practice becomes to pass a pointer to default constructed memory, then this will categorically increase branching cost. This would happen, for instance, if the return-by-value function is heavily used. 

Is the general idea to use this `not_null` version with properly initialized memory as far as possible?

Reply:
Makes sense on both accounts. changed to use `UNLIKELY`.